registry/test/apps.js

```diff
@@ -0,0 +1,388 @@
+        cssBundle: 'http://127.0.0.1:1234/ncTestAppReactssrUpdated.css',
+        ssr: {
+            src: "http://127.0.0.1:1234/fragmentUpdated",
+            timeout: 2000,
+            primary: false,
+        },
+        assetsDiscoveryUrl: 'http://127.0.0.1:1234/_spa/dev/assets-discoveryUpdated',
+        dependencies: {
+            react: 'https://cdnjs.cloudflare.com/ajax/libs/react/16.8.6/umd/react.development.js',
+            'react-dom': 'https://cdnjs.cloudflare.com/ajax/libs/react-dom/16.8.6/umd/react-dom.development.js',
+        },
+        props: {
+            fragmentModuleName: 'reactssr-app-mainUpdated',
+            assetsPath: 'http://127.0.0.1:3001/uisamplereactUpdated',
+            locationStrategy: 'browserHistoryUpdated',
+        },
+        initProps: {
+            assetsPath: 'http://127.0.0.1:3001/uisamplereact',
+        },
+    }),
+
+    create: async () => {
+        await request.post(example.url).send(example.correct);
+    },
+    delete: async () => {
+        await request.delete(example.url + example.encodedName);
+    },
+};
+example.encodedName = encodeURIComponent(example.correct.name);
+
+describe(`Tests ${example.url}`, () => {
+    before('should work simple "create" and "delete"', async () => {
+        await request.post(example.url).send(example.correct).expect(200)
+        await request.get(example.url + example.encodedName).expect(200);
+        await request.delete(example.url + example.encodedName).expect(204);
+        await request.get(example.url + example.encodedName).expect(404);
+    });
+    afterEach(example.delete);
+    describe('Create', () => {
+        it('should not create record without a required field: name', async () => {
+            const response = await request.post(example.url)
+            .send(_.omit(example.correct, 'name'))
+            .expect(422, '"name" is required');
+
+            expect(response.body).deep.equal({});
+        });
+
+        it('should not create record without a required field: spaBundle', async () => {
+            const response = await request.post(example.url)
+            .send(_.omit(example.correct, 'spaBundle'))
+            .expect(422, '"spaBundle" is required');
+
+            expect(response.body).deep.equal({});
+        });
+
+        it('should not create record with incorrect type of field: name', async () => {
+            let response = await request.post(example.url)
+            .send({
+                ...example.correct,
+                name: example.incorrect.name
+            })
+            .expect(422, '"name" must be a string');
+
+            expect(response.body).deep.equal({});
+
+            response = await request.get(example.url + example.incorrect.name)
+            .expect(404, 'Not found');
+
+            expect(response.body).deep.equal({});
+        });
+
+        it('should not create record with incorrect type of field: spaBundle', async () => {
```

Comment:
I guess we can create a test suite with the checking of validation
So, for example, you can have `when a user sends incorrect data` is a name of the test suite
Well a test case can have a name `should send a response with a validation error` or different
So the second test suite `when a user sends correct data` with a test case for example `should create an app` etc.